### PR TITLE
Improve ICA tests

### PR DESCRIPTION
**Pull request by ljoss17**
Closes: #3353

## Description

______

### PR author checklist:
- [x] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [x] Added tests: integration (for Hermes) or unit/mock tests (for modules).
- [x] Linked to GitHub issue.
- [x] Updated code comments and documentation (e.g., `docs/`).
- [x] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: .changelog/unreleased/improvements/ibc-integration-test/3353-improve-ica-tests.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+- Update ICA tests in order to use using `simd` instead of
+  `interchain-accounts-demo`.
```
Comment (romac) [inline .changelog/unreleased/improvements/ibc-integration-test/3353-improve-ica-tests.md:2]:
```suggestion
- Update ICA tests to use ibc-go's `simd` instead of `interchain-accounts-demo`.
```

File: tools/integration-test/Cargo.toml [hunk 2]
```diff
@@ -17,6 +17,7 @@ ibc-relayer-types  = { path = "../../crates/relayer-types" }
 ibc-relayer        = { path = "../../crates/relayer" }
 ibc-test-framework = { path = "../test-framework" }
 
+bincode = "1.0"
```
Comment (romac) [inline tools/integration-test/Cargo.toml:20]: Why is this needed? I cannot find a reference to it anywhere in the code.

Comment (ljoss17) [reply tools/integration-test/Cargo.toml:20]: Oh good catch! This is a leftover from an intermediary solution. I was serializing the `MsgSend` to a `Vec<u8>` inside the test using `bincode::serialize`, I will remove it.

**Review by romac — APPROVED** [review]
